### PR TITLE
refactor: conform to pyright strict

### PR DESCRIPTION
**Pull request by VincentRPS**
## Summary

Conforms the repository code for Pyright strict.

Supersedes #84

## PR Type

- [x] This PR improves the code (speed/style improvements or refactoring).
- [ ] This PR fixes an issue.
- [ ] This PR makes an addition to the library (new file, class, method, parameters, etc.).
- [ ] This PR makes a breaking change (renaming/removing parameters, methods, etc.).
- [ ] This PR doesn't make any code changes to the library (documentation, examples).

## Checklist

- [ ] I have searched all pull requests for duplicates.
- [ ] I have tested the changes made.
  - [ ] I have properly changed the docs to reflect the changes made.
- [ ] If 'type: ignore' comments were used, a comment is left to explain why it is needed.

### CODE REVIEW

File: pycord/api/route.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def __init__(
         ...
 
     def merge(self, url: str) -> str:
-        pass
```
Comment (EmreTech) [inline pycord/api/route.py:46]: How is using `...` instead of `pass` a Pyright Strict related change?

**Review by EmreTech — CHANGES_REQUESTED** [review]
A lot of the typing problems have been fixed by slapping on a `type: ignore` comment. This is hardly conforming to Pyright Strict.

**Comment by VincentRPS** [comment]
> A lot of the typing problems have been fixed by slapping on a `type: ignore` comment. This is hardly conforming to Pyright Strict.

Much of the errors are because of accessing private values on another class, of which I do not care about

File: pycord/api/route.py (hunk 1 above)
Comment (VincentRPS) [reply pycord/api/route.py:46]: It tells pyright it's a base class or something like that, because with `pass` it "returns" none but with `...` it's based

**Comment by EmreTech** [comment]
> > A lot of the typing problems have been fixed by slapping on a `type: ignore` comment. This is hardly conforming to Pyright Strict.
> 
> Much of the errors are because of accessing private values on another class, of which I do not care about

Then disable reports of private members being accessed?

**Pull request closed without merging**